src/lib.rs

```diff
@@ -14,6 +14,7 @@
 //! [`Arbitrary`](./trait.Arbitrary.html) trait's documentation for details on
 //! automatically deriving, implementing, and/or using the trait.
 
+#![cfg_attr(not(any(feature = "std", test)), no_std)]
```

Comment:
To avoid dealing with different preludes that are vs aren't implicitly in scope depending on the features enabled, we should make this crate *always* be `no_std`. And then because the `std` feature implies the `alloc` feature, we can layer in imports like

```rust
#[cfg(feature = "alloc")]
use alloc::boxed::Box;
```

and we don't have to use any complex logic like `all(feature = "alloc", not(feature = "std"))` kinds of things.

My experience has also been that tools like `rust-analyzer` handle this pattern for `no_std` crates better as well.